openmoc/process.py

```diff
@@ -595,3 +592,57 @@ def restore_simulation_state(filename='simulation-state.h5',
               '*.h5, *.hdf5, and *.pkl files are supported', filename)
 
     return {}
+
+
+##
+# @brief Parse an OpenMOC log file to obtain a simulation's convergence data.
+# @details This method compiles the eigenvalue and source residuals from each
+#          iteration of an OpenMOC simulation. This data is inserted into a 
+#          Python dictionary under the key names 'eigenvalues' and 'residuals',
+#          along with an integer `# iters`, and returned to the user.
+#
+#          This method may be called from Python as follows:
+# @code
+#          parse_convergence_data(filename='openmoc-XX-XX-XXXX--XX:XX:XX.log')
+# @endcode
+#
+# @param filename the OpenMOC log filename string
+# @param directory the directory where to find the log file
+# @return a Python dictionary of key/value pairs for convergence data
+def parse_convergence_data(filename, directory=''):
+
+  # If the user specified a directory
+  if len(directory) > 0:
+    filename = directory + '/' + filename
+
+  if not os.path.isfile(filename):
+    py_printf('ERROR', 'Unable to parse convergence data since "{0}" is ' + \
+              'not an existing OpenMOC log file'.format(filename))
+
+  # Compile regular expressions to find the residual and eigenvalue data
+  res = re.compile('res = ([0-9].[0-9]+E[+|-][0-9]+)')
+  keff = re.compile('k_eff = ([0-9]+.[0-9]+)')
+
+  # Parse the eigenvalues
+  with open(filename, 'r+') as f:
+    data = mmap.mmap(f.fileno(), 0)
+    eigenvalues = keff.findall(data)
```

Comment:
Thanks for the comment - I'll look into it. I should probably start using Python 3 exclusively for my work to avoid these sorts of issues.